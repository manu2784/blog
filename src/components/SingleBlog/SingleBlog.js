import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
    CardHeader,
    CardMedia,
    CardContent,
    CardActions,
    Avatar,
    IconButton,
    Typography,
    Button
} from "@material-ui/core";
import { red } from "@material-ui/core/colors";
import FavoriteIcon from "@material-ui/icons/Favorite";
import ShareIcon from "@material-ui/icons/Share";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import ShowMore from 'react-show-more';
import ThumbUpIcon from '@material-ui/icons/ThumbUp';

const useStyles = makeStyles(theme => ({
  root: {
    width:"70%",
    margin: "15px 0px"
  },
  media: {
    height: 0,
    paddingTop: "56.25%" // 16:9
  },
  expand: {
    transform: "rotate(0deg)",
    marginLeft: "auto",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest
    })
  },
  expandOpen: {
    transform: "rotate(180deg)"
  },
  avatar: {
    backgroundColor: red[500]
  },
   vote:{
    fontSize:".8rem",
       margin:"0 4px"
   }
}));

function SingleBlog(props) {
  const classes = useStyles();

  function checkMedia(data) {
        let image = data.url.match(/\.(jpeg|jpg|gif|png)$/) != null;
        if(image) return "image";

        if(data.secure_media_embed.content)
          return "embeded_media";

        return "link";
    }
  return (
    <React.Fragment>
      {props.posts.map(post => (
        <Card className={classes.root} key={post.data.name}>
          <Button href={post.data.url} >
          <CardHeader
            avatar={
              <Avatar aria-label="recipe" className={classes.avatar}>
                  {post.data.author.charAt(0)}
              </Avatar>
            }
            title={post.data.title}
            subheader={new Date(post.data.created *1000).toISOString().split("T").join(" ").slice(0, -8)}
          /></Button>
            { checkMedia(post.data) === "image" && <CardMedia
            className={classes.media}
            image={post.data.url}
            title=""
          />}

            {checkMedia(post.data) === "embeded_media" &&
            <iframe src={post.data.secure_media_embed.media_domain_url} width={post.data.secure_media_embed.width} height={post.data.secure_media_embed.height}></iframe>
            }


          <CardContent>
            <ShowMore
                lines={3}
                more='Show more'
                less='Show less'
                anchorClass=''
            >
            <Typography variant="body2" color="textSecondary" component="p">
            {post.data.selftext}
            </Typography>
            </ShowMore>
          </CardContent>
          <CardActions disableSpacing>
            <IconButton aria-label=" vote score">
                <ThumbUpIcon /> <span className={classes.vote}> {post.data.score}</span>
            </IconButton>
            <IconButton aria-label="share">
              <ShareIcon />
            </IconButton>
          </CardActions>
        </Card>
      ))}
    </React.Fragment>
  );
}

export default SingleBlog;
