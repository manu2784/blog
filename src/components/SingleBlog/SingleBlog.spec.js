import React from "react";
import { shallow, configure, mount } from "enzyme";
import SingleBlog from "./SingleBlog";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



describe("SingleBlog", () => {

let posts = [{
    data: {
        selftext: "jdkhgf",
        url: "https://www.reddit.com/r/reactjs/",
        author: "fdskjhfkdj",
        created:"1580630711",
        secure_media_embed:{}
    }
}
];


    it("should render single blog component", () => {

        const blog = shallow(<SingleBlog posts={posts}/>);
        expect(blog).toMatchSnapshot();
    });
});