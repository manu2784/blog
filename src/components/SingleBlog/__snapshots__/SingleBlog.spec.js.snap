// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList should render single blog component 1`] = `ShallowWrapper {}`;

exports[`ImageGrid should render single blog component 1`] = `ShallowWrapper {}`;

exports[`ImageGrid should render the image grid page 1`] = `ShallowWrapper {}`;

exports[`SingleBlog should render single blog component 1`] = `ShallowWrapper {}`;
